design/secrets_provider_milestone1_design.md

```diff
@@ -0,0 +1,385 @@
+   ...
+      env:
+        - name: K8S_SECRET_LABELS
+          value: 'configurable-key:env-prod,configurable-key:env-test,'
+```
+
+These labels will have an OR inclusive relationship not an AND relationship. A customer can also write '*' to imply all K8s secrets in the namespace. If K8S_SECRET_LABELS is not supplied, by default, the Secrets Provider will fetch all K8s Secrets in the namespace.
+
+To read the labels set on K8s Secrets, the Service Account used for the Secrets Provider will need "list" privileges on K8s Secrets resources.
+
+The Service Account / Role / RoleBinding will be defined as the following:
+
+```yaml
+# Service Account definition 
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: conjur-sa-1
+
+# Role definition
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  namespace: app-namespace
+  name: secrets-access
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "update", "list"] # new privilege here
+
+# RoleBinding definition to associate the SA with the Role
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: secrets-access-binding
+  namespace: app-namespace
+subjects:
+  - kind: ServiceAccount
+    name: conjur-sa-1
+    apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: Role
+  name: secrets-access
+  apiGroup: rbac.authorization.k8s.io
+```
+
+#### Code changes
+
+Although we will now offer "K8S_SECRET_LABELS" for easy filtering on K8s Secrets, we will still support the "K8S_SECRET" environment variable if the customer prefers to list the K8s Secrets they need values for from Conjur. 
+
+
+Decision: **Solution #1, Read all K8s Secrets that are marked with labels**
+
+We decided against listing K8s Secrets as configuration in the ENV of the Secrets Provider manifest because that is not a scalable solution, especially for customers with hundreds of K8s Secrets. Customers might already use labels for grouping related K8s Secrets so we will benefit by integrating more with a customer environment and utilizing that.
+
+### Order of Deployment
+
+To ensure that on first run our Secrets Provider runs first, we will request from the customer follow the following setup order:
+
+1. Add all necessary K8s Secrets and their labels to the Secrets Provider manifest
+
+2. Run the Secrets Provider
+
+3. Run application pods
+
+### Lifecycle/Deletion
+
+The lifecycle of the Job is the length of the process. In other words, the time it takes to retrieve the Conjur secrets value and update them in the K8s Secrets. The customer can configure the Job to stay "alive" by adding **ttlSecondsAfterFinished**, with the desired about of seconds, to their Secrets Provider yaml. Because of the nature of a Job, a manual delete is not necessary.
+
+### Backwards compatibility
+
+Milestone 1 of Phase 2 will be built ontop of the current init container solution. We will delivery the same image and will not be erasing code and any current functionality.
```

Comment:
that is incorrect. we have a breaking change in the required privilege for the service account and we also need to refer to the change in the way that we get the required k8s secrets. please address these points in this section.